Migrate to ROS2 Humble with ROS Action support

MishkaMN opened the pull request:
<!-- Thanks for the contribution, this is awesome. -->

# PR Details
## Description
This PR brings our fork up to date with the ros2 repo that allows the ros1_bridge to build with all the autoware and carma messages. 
This PR also merges ROS action support from here https://github.com/smith-doug/ros1_bridge/tree/action_bridge_humble
This PR also updates all github action versions and removes `required-ros-distributions: "noetic rolling"` which is intended to install the ROS versions from distribution because ROS1 noetic is not available on 22.04 anymore. By removing it, the `setup-ros` action installs it from source. 
<!--- Describe your changes in detail -->

## Related GitHub Issue
NA
<!--- This project only accepts pull requests related to open GitHub issues or Jira Keys -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please DO NOT name partially fixed issues, instead open an issue specific to this fix -->
<!--- Please link to the issue here: -->

## Related Jira Key
[ARC-156](https://usdot-carma.atlassian.net/browse/ARC-156)
<!-- e.g. CAR-123 -->

## Motivation and Context
Upgrade to ROS2 Humble
<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?
Built using dockerhub using carma-msgs Dockerifle in https://github.com/usdot-fhwa-stol/carma-msgs/pull/248
And tested in cdasim xil carma cloud use case 
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Defect fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [X] I have added any new packages to the sonar-scanner.properties file
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.


[ARC-156]: https://usdot-carma.atlassian.net/browse/ARC-156?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

hsd-dev committed:
added action templates

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
generate action interface mappings

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
added action factories

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
added action_bridge executable

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
fix template

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
hack for GripperCommand

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
Update include/ros1_bridge/action_factory.hpp

Co-authored-by: tomoya <Tomoya.Fujita@sony.com>
Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

Commit:
Fix python script errors in action generation

Signed-off-by: Victor Lopez <victor.lopez@pal-robotics.com>
Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

Commit:
Foxy and later action compatibility

Signed-off-by: Victor Lopez <victor.lopez@pal-robotics.com>
Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

Commit:
Reorder ros node initialization to allow renaming of bridge

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
Update resource/interface_factories.cpp.em

Co-authored-by: Victor Lopez <3469405+v-lopez@users.noreply.github.com>
Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
applied uncrustify patches

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
fix action mapping rules

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
print available action pairs

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
get active ROS1 action servers and clients

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
remove prints

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
fix mapping rules for services

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
fix error during rebase

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
updated deprecated GoalResponseCallback signature

https://github.com/ros2/rclcpp/blob/master/rclcpp_action/include/rclcpp_action/client.hpp#L289-L299

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
reformatted with uncrustify

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
map actions in dynamic_bridge

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
shutdown internal server before removing the bridge

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
updated GoalResponseCallback signature

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
updated README

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
fix formatting

added NOLINT whenever both cpplint and uncrustify cannot be satisfied
https://github.com/ament/ament_lint/issues/158

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

Blast545 committed:
Fix cpplint error (#341)

* Fix cpplint error

Signed-off-by: Jorge Perez <jjperez@ekumenlabs.com>

nirwester committed:
Parametrizing service execution timeout (#340)

* Parametrizing service execution timeout

Signed-off-by: Marco Bassa <marco@robotcloud.eu>

Timple committed:
The service name might provide more debug information than host:port details (#346)

* The service name might provide more debug information than host:port details

Signed-off-by: Tim Clephas <tim.clephas@nobleo.nl>
Co-authored-by: Tomoya Fujita <Tomoya.Fujita@sony.com>

ckaran committed:
Cleanup of README.md (#342)

* Reformatted README.md

This is an initial cleanup attempt of the README.md file.  All
non-code lines have been wrapped to an 80 column width, and some
markdown was cleaned up.  Further work is needed.

Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* Added 'bash' info string to appropriate fenced code blocks.

Numerous fenced code blocks did not have a
[info string](https://github.github.com/gfm/#info-string), which
meant that the enclosed code block wouldn't be rendered correctly.
As it turned out, all of these were intended to be run in a bash
shell, so I added that as their info strings.

Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* WIP: Updated the scripts in the README.md file to use environment variables.

The scripts weren't directly executable before, requiring manual
substitution of various 'variables'.  This was unfortunate as Github
makes copying and pasting code blocks simple and straightforward.

The changes in this commit make it possible to copy/paste each example
into it's own shell, which should make it slightly easier for new users
to start using the bridge.

Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* fix: Unwrapped lines per request from @gbiggs.

@gbiggs commented at https://github.com/ros2/ros1_bridge/pull/342#issuecomment-1029508012
that files in ROS2 are intentionally not wrapped to make handling
diffs in PRs easier.  I've copied in the relevant original lines
to unwrap the lines I wrapped earlier.

Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* Update README.md

Accepted suggestion from @gbiggs

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* Update README.md

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* Update README.md

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* Update README.md

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* Update README.md

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* Update README.md

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* Update README.md

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* Update README.md

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* Update README.md

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* The service name might provide more debug information than host:port details (#346)

* The service name might provide more debug information than host:port details

Signed-off-by: Tim Clephas <tim.clephas@nobleo.nl>
Co-authored-by: Tomoya Fujita <Tomoya.Fujita@sony.com>
Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

* fix: Changed 'galactic' to 'rolling'

Changes due to suggestions from @gbiggs in the discussion of [PR 342](https://github.com/ros2/ros1_bridge/pull/342#discussion_r834287510)

Signed-off-by: Cem Karan <cem.f.karan.civ@army.mil>

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Co-authored-by: Tim Clephas <tim.clephas@nobleo.nl>
Co-authored-by: Tomoya Fujita <Tomoya.Fujita@sony.com>

gbiggs committed:
Changelog

Signed-off-by: Geoffrey Biggs <gbiggs@killbots.net>

gbiggs committed:
0.10.3

clalancette committed:
Make sure to use fully-qualified boost placeholders. (#351)

Also make sure to declare a direct dependency on boost.
While we are always getting it from ROS 1, this code directly
uses it and so should also declare a dependency.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

Commit:
Suppress warning when packages aren't available (#355)

* Suppress warning when packages aren't available

ROS1 packages aren't available on all platforms.  In order to keep
ros1_bridge in ros2.repos without warnings turning CI yellow, we need to
suppress this warning.

Signed-off-by: Michael Carroll <michael@openrobotics.org>

methylDragon committed:
Update maintainers (#360)

Signed-off-by: methylDragon <methylDragon@gmail.com>

quarkytale committed:
updating maintainers

Signed-off-by: quarkytale <dharini@openrobotics.org>

quarkytale committed:
reverting authors

Signed-off-by: quarkytale <dharini@openrobotics.org>

quarkytale committed:
Merge pull request #361 from ros2/quarkytale/maintainers

Update maintainers

xlla committed:
to correct Examples 4 (#357)

Commit:
Fix array static_assertion with newer compilers (#363)

Signed-off-by: Michael Carroll <michael@openrobotics.org>

methylDragon committed:
Clarify example 4 (#362)

Signed-off-by: methylDragon <methylDragon@gmail.com>

quarkytale committed:
added fields_2_to_1 to print message mapping

Signed-off-by: Dharini Dutia <dharini@openrobotics.org>

quarkytale committed:
fixed ros2 message index

Signed-off-by: Dharini Dutia <dharini@openrobotics.org>

quarkytale committed:
updated description

Signed-off-by: Dharini Dutia <dharini@openrobotics.org>

hsd-dev committed:
add actionlib dependency with condition

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

quarkytale committed:
Merge pull request #366 from ros2/quarkytale/parsing_ros2_msg_fields

Fixed index for ros2 namespace type messages

quarkytale committed:
Merge branch 'master' into quarkytale/mapping_rule

LoyVanBeek committed:
Parametrize Quality of Service in `parameter_bridge`.  (#331)

* Extend create_bidirectional_bridge to take qos param for ROS2 publisher

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Busy setting up a way to read QoS parameters from ROS1 params

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Parse history qos params

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Call qos_from_params when setting up topics

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Configure deadline, lifespan, liveliness_lease_durations

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Configure liveliness

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Add some basic debug text

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Print the QoS settings to stdout when setting them up

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Catch XmlRpc::XmlRpcExceptions when constructing QoS from parameters

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Parse liveliness as either int enum value or upper/lower case string representation of liveliness enum values

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Fix formatting with uncrustify

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Fix cpplint formatting

In 2 cases the formatting conflicts with what uncrustify wants; // NOLINT got rid of the complaints of cpplint, favouring uncrustify

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Clearer logging as suggested by code review

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Clarify keep_last vs keep_all setting for history

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>

quarkytale committed:
removing unused slot

Signed-off-by: Dharini Dutia <dharini@openrobotics.org>

quarkytale committed:
Merge pull request #365 from ros2/quarkytale/mapping_rule

Print out message mapping for fields_2_to_1 rule

methylDragon committed:
Implement Foreign Mapping Check Override (#367)

hsd-dev committed:
added dependency on rclcpp_action

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

gbiggs committed:
Move xmlrpcpp find_package so it only searches if ROS 1 is found (#371)

Signed-off-by: Geoffrey Biggs <gbiggs@killbots.net>

AndrejOrsula committed:
Fix typo (`services_1_or_2` -> `services_1_to_2`) (#379)

Signed-off-by: Andrej Orsula <orsula.andrej@gmail.com>

Signed-off-by: Andrej Orsula <orsula.andrej@gmail.com>

dbking77 committed:
New function for converting between ROS1 and ROS2 generic message formats. (#377)

Signed-off-by: Derek King <derek.king@zebra.com>

Signed-off-by: Derek King <derek.king@zebra.com>

LoyVanBeek committed:
Apply automatic mapping rules in case only package+message mapping exists (#382)

* Fix message mapping by removing early return so other rules can still be applied

In determine_field_mapping, there was an early return inside a loop over all mapping rules.
IF there we any mapping rules but they don't specify field mappings, the early return made the function return without creating mappings automatically.

For a particular message type, ROS 1's uuid_msgs/UniqueID vs ROS 2's unique_identifier_msgs/UUID, the message definition is exacly the same but type name is not.
The only mapping fule defined in for unique_identifier_msgs/UUID is that it maps to uuid_msgs/UniqueID, but no field mappings are needed because the definitions are the same.

But, then we hit the early return (because the for-loop is ran without any rule applying to the message at hand and thus not `continue`-ing in a code branch handling a rule)
and return without applying the normal automatic field mapping generation rules.

By removing the early return, the other rules are applied and the mapping rules for handling the exact same message defintions are applied

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Account for fields mapped by rules when checking for missed fields

The code after the early return mentioned in the previous commit assumed all fields would match by name,
which was of course true. But not anymore, so the missing check now only fails when the missing fields are also not already accounted for via a mapping

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

* Fix flake8 violations

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

Signed-off-by: Loy van Beek <loy.vanbeek@mojin-robotics.de>

audrow committed:
[master] Update maintainers - 2022-11-07 (#386)

hsd-dev committed:
Update CMakeLists.txt

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
Update README.md

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
Update include/ros1_bridge/action_factory.hpp

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
Update resource/get_factory.cpp.em

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
Update ros1_bridge/__init__.py

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
Update src/action_bridge.cpp

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>
Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

dbking77 committed:
Provide direct serialization of ROS2 messsage to ROS1 streams (#381)

* Generated functions for (de)serializing ROS2 messages to ROS1 streams.

Signed-off-by: Derek King <derek.king@zebra.com>

* Partial set of fixes from review.

Signed-off-by: Derek King <derek.king@zebra.com>

* Rename functions based on discussion from review

Signed-off-by: Derek King <derek.king@zebra.com>

* Whitespace fix.

Signed-off-by: Derek King <derek.king@zebra.com>

* Name change for conversion helper fucntion.

Signed-off-by: Derek King <derek.king@zebra.com>

* Use explicit function overloading instead of template for stream types.

Signed-off-by: Derek King <derek.king@zebra.com>

* Fix factory.hpp crustify errors.

Signed-off-by: Derek King <derek.king@zebra.com>

Signed-off-by: Derek King <derek.king@zebra.com>

Nilaos committed:
Improved gitignore to exclude build files and similar editor-based tools (#392)

Signed-off-by: Nicholas Sims <nicholas.d.sims@gmail.com>

Signed-off-by: Nicholas Sims <nicholas.d.sims@gmail.com>

hsd-dev committed:
risk leaking if the reset call crashes somehow

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
Add goal ID to log output so it's more useful

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
rename variable so it's distinction from the variable with the same name in the enclosing class

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
Change cout to use RCLCPP logging macros.

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
remove commented out code

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
fix parenthesis

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
remove unused params

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
fix nullptr keyword

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
remove outdated comments

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

hsd-dev committed:
pass logger to GoalHandler

Signed-off-by: Harsh Deshpande <harshavardhan.deshpande@ipa.fraunhofer.de>

finostro committed:
FI: fixes bug when an action has a field that matches the subfield (eg a goal has a field named goal , or a feedback has a feedback in it)

Signed-off-by:  Douglas Smith <douglas.smith@swri.org>

oKermorgant committed:
handle special case of std::vector<bool>

Signed-off-by: Olivier Kermorgant <olivier_kermorgant@yahoo.fr>

oKermorgant committed:
simplify srv template to allow vector<bool>

Signed-off-by: Olivier Kermorgant <olivier_kermorgant@yahoo.fr>

smith-doug committed:
Use humble action goal_response_callback signature when ros2 versions is humble/iron

Signed-off-by:  Douglas Smith <douglas.smith@swri.org>

smith-doug committed:
Merge remote-tracking branch 'oKermorgant/master' into action_bridge_humble

Signed-off-by:  Douglas Smith <douglas.smith@swri.org>

smith-doug committed:
Make sure action rules don't get applied in get_ros2_messages, get_ros2_services

Signed-off-by:  Douglas Smith <douglas.smith@swri.org>

smith-doug committed:
Support ros1_bridge_foreign_mapping in get_ros2_actions

Signed-off-by:  Douglas Smith <douglas.smith@swri.org>

smith-doug committed:
Accidentally made automatic mapping of actions when the ros2 action has more stuff like FollowJointTrajectory

Signed-off-by:  Douglas Smith <douglas.smith@swri.org>

smith-doug committed:
Use rosout qos for /rosout in dynamic_bridge

Signed-off-by:  Douglas Smith <douglas.smith@swri.org>

smith-doug committed:
Optionally allow missing fields in message mapping

Signed-off-by:  Douglas Smith <douglas.smith@swri.org>

smith-doug committed:
Remove unneeded None checks, add exception catch in missing field check

Signed-off-by:  Douglas Smith <douglas.smith@swri.org>

smith-doug committed:
allow_missing_fields defaults to false

My intent was to allow things like Log messages to map if defined via mapping_rules yaml.  This didn't work with Log because the defines for the different levels changed, so rqt's console display would go crazy about invalid severity.  If this is useful, it will have to be added to the yaml rules definition to be exposed.

Signed-off-by:  Douglas Smith <douglas.smith@swri.org>

smith-doug committed:
Add Log message conversion

Putting it in builtin interfaces for now.

Signed-off-by:  Douglas Smith <douglas.smith@swri.org>

MishkaMN committed:
boost::placeholders::_1

MishkaMN committed:
add const

MishkaMN committed:
add more const

MishkaMN committed:
failing place

MishkaMN committed:
Merge branch 'master' of https://github.com/ros2/ros1_bridge into develop-humble-buildable

MishkaMN committed:
reorder default paramters

MishkaMN committed:
remove redefinition

MishkaMN committed:
read/write for vector bool

MishkaMN committed:
use the vectorbool

MishkaMN committed:
Fix unit test

MishkaMN committed:
further fix for default parameter

MishkaMN committed:
Merge branch 'action_bridge_humble' of https://github.com/smith-doug/ros1_bridge into develop-humble-buildable

john-chrosniak commented on CODEOWNERS:
```diff
@@ -0,0 +1,2 @@
+# This file was generated by https://github.com/audrow/update-ros2-repos
+* @methylDragon @quarkytale @gbiggs
```
It looks like github is complaining since these developers don't have write access to our repo, could we remove this?

MishkaMN replied:
yeah we can

MishkaMN replied:
yeah we can. removed.

john-chrosniak commented:
It looks like CI is failing when running `sudo apt-get update` because it is looking for a ROS1 installation on 22.04. Is that due to [carma-msgs #248](https://github.com/usdot-fhwa-stol/carma-msgs/pull/248) not being merged yet?

MishkaMN committed:
Delete CODEOWNERS

MishkaMN commented:
> It looks like CI is failing when running `sudo apt-get update` because it is looking for a ROS1 installation on 22.04. Is that due to [carma-msgs #248](https://github.com/usdot-fhwa-stol/carma-msgs/pull/248) not being merged yet?

Looks like the Github Action versions were outdated. I updated all github action versions and removed `required-ros-distributions: "noetic rolling"` which is intended to install the ROS versions from distribution because ROS1 noetic is not available on 22.04 anymore. By removing it, the `setup-ros` action installs it from source.

MishkaMN committed:
Update main.yml to try make CI work

MishkaMN committed:
merge github action upgrades (#28)

<!-- Thanks for the contribution, this is awesome. -->

# PR Details
## Description

<!--- Describe your changes in detail -->

## Related GitHub Issue

<!--- This project only accepts pull requests related to open GitHub
issues or Jira Keys -->
<!--- If suggesting a new feature or change, please discuss it in an
issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps
to reproduce -->
<!--- Please DO NOT name partially fixed issues, instead open an issue
specific to this fix -->
<!--- Please link to the issue here: -->

## Related Jira Key

<!-- e.g. CAR-123 -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran
to -->
<!--- see how your change affects other areas of the code, etc. -->

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all
the boxes that apply: -->

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality
to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes
that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're
here to help! -->

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the
[**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md)
document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

MishkaMN committed:
Update main.yml
